### PR TITLE
Add /calculate_pp and improve performance usecase

### PR DESCRIPTION
**Pull request by minisbett**
Edit: Accidentally put the GDPR stuff in here so I guess merge #341 before this

Added the /calculate_pp endpoint. It takes the api key in the Authorization header.

In the query parameters, beatmap_id is the only required parameter. mods, mode, misses etc. all have a default value of 0, combo has the maximum combo of the beatmap as the default.

The return type of the endpoint is a single JSON object if the n100 and n50 parameters were used. If the acc parameter is used instead, the user can specify multiple accuracies in the format "99.1,98.38,88.23". In that case, a list of JSON objects is returned.
While changing return types on API endpoints are usually bad, we can do it here because the developer explicitly either requests a singlg object via n100 and n50 or multiple via the acc parameter. Therefore the developer can expect what the return type is.

The extra n100 and n50 is necessary as the pp system has a 50 hit penalty. If you specify accuracies, rosu-pp automatically assumes the best-case-scenario for the 50 count. This is usually desired when requesting a list of accuracies, e.g. for visualizing the pp values for common accuracies (100, 99, 98, 95, ...)

I think adding the possibility of specfying multiple n100 and n50 pairs would bloat the endpoint with parsing and does not really have much real world applications. And since the performance usecase does not optimize for calculating multiple scores anyways (e.g. by caching star ratings) the user could just send multiple requests to the endpoint if more than one n100/n50 pair is necessary.

I also reworked the return of the performance usecase and adapted the code that uses the calculate_performances() method to it. It now returns all performance and difficulty-related values instead of just the pp and star rating.

### CODE REVIEW

File: app/commands.py [hunk 1]
```diff
@@ -164,6 +165,18 @@ def wrapper(f: Callback) -> Callback:
...
+async def gdpr(ctx: Context) -> Optional[str]:
+    code = str(uuid.uuid4())
+
+    for _code, (user_id) in dict(app.state.sessions.gdpr_codes).items():
+        if user_id is ctx.player.id:
+            del app.state.sessions.gdpr_codes[_code]
+
+    app.state.sessions.gdpr_codes[code] = (ctx.player.id, time.time() + 3) # code expires after 5 minutes
+
+    return f"Your GDPR data is available here: https://api.{app.settings.DOMAIN}/get_gdpr_data?code={code}\nThe link expires in 5 minutes."
```
[the remark this answers is not in the document]
Comment (minisbett) [reply app/commands.py:186]: actually yes

[the remark this answers is not in the document]
Comment (minisbett) [reply app/commands.py:186]: wait

[the remark this answers is not in the document]
Comment (minisbett) [reply app/commands.py:186]: OH FFS

[the remark this answers is not in the document]
Comment (minisbett) [reply app/commands.py:186]: It also put the gdpr shit in this PR

**Comment by NiceAesth** [comment]
This PR should be rebased to master and force pushed to remove data dumping functionality. I myself do not approve of this PR otherwise as it would depend on a completely unrelated PR which is not even guaranteed for merging. @cmyui would like your thoughts on this as well.

**Comment by minisbett** [comment]
Yes I fucked up the git history as said in the PR above, can you come to #programming and help me fix it?

**Pull request closed without merging**